plugins/macro/Macro/Utilities.pm

```diff
@@ -49,7 +49,7 @@ sub ai_isIdle {
 		}
 	}
 	return 1 if (AI::is('macro', 'deal'));
-	return 1 if (AI::is('NPC') && $char->args->waitingForSteps);
+	return 1 if (AI::is('NPC') && $char->args->{waitingForSteps});

```

Comment:
This shouldn't ever happen. `$char->args` is really `AI::args`, and the `AI::args` for an NPC AI sequence should always be a `Task::TalkNPC`, which has `waitingForSteps`. Whatever code is creating the need for this change should be fixed instead.

Reply:
I had a problem with an old macro that use do autosell do you think this will fix this problem when talk to the vender npc? @alisonrag

Reply:
I do like to inform that this PR corrected the issue from macro.pl autosell conversation: before this my macro was not finishing the autosell and now it is: 
![image](https://user-images.githubusercontent.com/17572686/29744746-d68abfbc-8a81-11e7-9e85-697ce4469fd7.png)